numpy/core/tests/test_umath_accuracy.py

```diff
@@ -5,13 +5,15 @@
 import pytest
 from ctypes import c_float, c_int, cast, pointer, POINTER
 from numpy.testing import assert_array_max_ulp
+from numpy.core._multiarray_umath import __cpu_features__
 
-runtest = sys.platform.startswith('linux') and (platform.machine() == 'x86_64')
+IS_AVX = __cpu_features__.get('AVX512F') or \
+        (__cpu_features__.get('FMA3') and __cpu_features__.get('AVX2'))
+runtest = sys.platform.startswith('linux') and IS_AVX
```

Comment:
You could write this as `runtest = __cpu_features__.get('AVX512F', False) or (__cpu_features__.get('FMA3', False) and __cpu_features__.get('AVX2', False))`

Reply:
hmm. Now you don't need the sys.platform (someone brave might try to compile on windows)

Reply:
It could still call the C standard library of that platform even if AVX is detected at runtime. I know for sure glibc passes these tests on linux, I haven't tested on other OS. Hence the linux constraint.

Reply:
Question about `__cpu_features__`. Does it detect the available CPU features, or did it only find the CPU features that are both available *and* targeted in the build? I assume the "and targeted in the build", or should we make a distinction?

Reply:
I think it only detects the available CPU features and has nothing to do with what is targeted in the build.

Reply:
I will create an issue to resolve that one, knowing what NumPy is targeted for might be nice, and actually more useful here, no?

Reply:
umm, won't build targets depend on each Ufunc's implementation? For example, float32 sin/cos/exp/log are build for AVX2 and AVX512F but float64 versions are not.

Reply:
But when you build NumPy you build for a maximal version? I.e. my computer has no AVX2 and AVX512F, so if I build the binary without that and then copy it to a server which has both, my NumPy will definitely not use them. So even though I will not use them, should `__cpu_features__` report them?

Reply:
As far as I understand, the current build system does not depend on the CPU you build NumPy on.  It only depends on the compiler you use. As long as the compiler can generate AVX2/AVX512F instructions, the Ufunc is build for AVX512F, AVX2 and regular C (assuming the AVX versions exist in the source code). This is done via the function attributes.

So, in your example, if you build on a non AVX machine and copy the binary over to a AVX enabled machine, the runtime detects AVX512 and it will run the AVX512 version of the Ufunc.

Reply:
Right, but I still can choose to limit myself (i.e. because my computer cannot run it anyway). And for diagnostics (such as in this tests) it does not really matter what the system supports, what matters whether the running instance is making use of the feature.

Reply:
Ah, I see what you mean. That would mean having both HAVE_ATTRIBUTE_TARGET_AVX512F_WITH_INTRINSICS (which is used for building) and runtime detected features available in the dictionary. I think that would be useful.